src/xhydro/frequency_analysis/regional.py

```diff
@@ -0,0 +1,702 @@
+        ds_out[v].attrs["Z_threshold"] = thresh_z
+        ds_out[v].attrs["description"] = "Mask for regions based on H & Z thresholds"
+        ds_out[v].attrs["history"] = utils.update_history(
+            f"Mask for regions based on H ({thresh_h}) & Z ({thresh_z}) thresholds",
+            ds_out[v],
+        )
+    return ds_out
+
+
+def _combine_h_z(ds: xr.Dataset) -> xr.Dataset:
+    new_ds = xr.Dataset()
+    for v in ds:
+        if "_Z" in v:
+            new_ds[v.removesuffix("_Z")] = xr.concat(
+                [ds[v.replace("_Z", "_H")], ds[v]], dim="crit"
+            ).assign_coords(crit=["H", "Z"])
+    return new_ds
+
+
+def calculate_rp_from_afr(
+    ds_groups: xr.Dataset,
+    ds_moments_groups: xr.Dataset,
+    rp: np.array,
+    l1: xr.DataArray | None = None,
+) -> xr.DataArray:
+    """
+    Calculate return periods from Annual Flow Regime (AFR) analysis.
+
+    Parameters
+    ----------
+    ds_groups : xarray.Dataset
+        Dataset containing grouped flow data.
+    ds_moments_groups : xarray.Dataset
+        Dataset containing L-moments for grouped data.
+    rp : array-like
+        Return periods to calculate.
+    l1 : xarray.DataArray, optional
+        First L-moment (location) values. L-moment can be specified for ungauged catchments.
+        If None, values are taken from ds_moments_groups.
+
+    Returns
+    -------
+    xarray.DataArray
+        Calculated return periods for each group and specified return period.
+
+    Notes
+    -----
+    This function calculates return periods using the Annual Flow Regime method.
+    If l1 is not provided, it uses the first L-moment from ds_moments_groups.
+    The function internally calls calculate_ic_from_AFR to compute the flood index.
+    Equations are based on Hosking, J. R. M., & Wallis, J. R. (1997). Regional frequency analysis (p. 240).
+    """
+    if l1 is None:
+        station_dim = ds_moments_groups.cf.cf_roles["timeseries_id"][0]
+        l1 = ds_moments_groups.sel(lmom="l1").dropna(dim=station_dim, how="all")
+    return _calculate_ic_from_afr(ds_groups, ds_moments_groups, rp) * l1
+
+
+def _calculate_ic_from_afr(
+    ds_groups: xr.Dataset, ds_moments_groups: xr.Dataset, rp: list
+) -> xr.Dataset:
+
+    lambda_r_1, lambda_r_2, lambda_r_3 = _calc_lambda_r(ds_groups, ds_moments_groups)
+
+    # alpha = location
+    # xi    = scale
+    # kappa = shape
+
+    kappa = _calc_kappa(lambda_r_2, lambda_r_3)
+
+    term = xr.apply_ufunc(_calc_gamma, (1 + kappa), vectorize=True)
+
+    # Hosking et Wallis, eq. A56. et Anctil et al. 1998, eq. 7 et 8.
+    alpha = (lambda_r_2 * kappa) / ((1 - (2**-kappa)) * term)
+    xi = lambda_r_1 + (alpha * (term - 1)) / kappa
+
+    # Calculating wanted return periods
+    t = xr.DataArray(data=rp, dims="return_period").assign_coords(return_period=rp)
+
+    # Hosking et Wallis, eq. A44 et Anctil et al. 1998, eq. 5.
+    q_rt = xi + alpha * (1 - (-np.log((t - 1) / t)) ** kappa) / kappa
+
+    return q_rt
```

Comment:
Cette fonction devrait modifier les attributs des variables. Quelque chose comme:

["long_name"] = "Return period"
["description"] = "Calculated return periods for each group and specified return period."
["units"] = (juste être certain que c'est encore la bonne chose)